ui/src/app/views/dataset/component.tsx

```diff
@@ -58,15 +92,102 @@ export const DataSet = withCurrentWorkspace()(class extends React.Component<
       const {namespace, id} = this.props.workspace;
       const conceptSets = await Promise.resolve(conceptSetsApi()
         .getConceptSetsInWorkspace(namespace, id));
-      this.setState({conceptSetList: conceptSets.items});
+      this.setState({conceptSetList: conceptSets.items, loadingConceptSets: false});
     } catch (error) {
       console.log(error);
     }
   }
 
+  convertResource(r: ConceptSet | Cohort, rType: ResourceType): RecentResource {
+    const {workspace} = this.props;
+    this.setState({rType: rType});
+    console.log(rType);
+    return convertToResource(r, workspace.namespace, workspace.id,
+      workspace.accessLevel as unknown as WorkspaceAccessLevel, rType);
+  }
+
+  openConfirmDelete(r: ConceptSet, rType: ResourceType): void {
+    const rc = this.convertResource(r, rType);
+    this.setState({confirmDeleting: true, resource: rc});
+  }
+
+  closeConfirmDelete(): void {
+    this.setState({confirmDeleting: false});
+    if (this.state.rType === ResourceType.CONCEPT_SET) {
+      const deleted = this.state.resource.conceptSet;
+      const updatedList = fp.filter(conceptSet => conceptSet.id !== deleted.id,
```

Comment:
Interesting question on this one (I don't know the answer): On deleting, should we just filter out of the list, or reload the conceptSetList? I don't know what we do elsewhere, but we should be consistent. (Tradeoff being that this is faster, but means a user is more likely to be left with a stale list.)

Reply:
This is the way I'd _like_ to have it implemented. It's much faster, and since we have etags to prevent updates to stale workspaces I feel like it's safe.